internal/testcontainersdocker/docker_socket.go

```diff
@@ -11,3 +18,24 @@ var DockerSocketPathWithSchema = DockerSocketSchema + DockerSocketPath
 
 // TCPSchema is the tcp schema.
 var TCPSchema = "tcp://"
+
+func init() {
+	const DefaultDockerHost = client.DefaultDockerHost
+
+	u, err := url.Parse(DefaultDockerHost)
+	if err != nil {
+		// unsupported default host specified by the docker client package,
+		// so revert to the default unix docker socket path
+		return
+	}
+
+	switch u.Scheme {
+	case "unix", "npipe":
+		DockerSocketSchema = u.Scheme + "://"
+		DockerSocketPath = u.Path
+		if !strings.HasPrefix(DockerSocketPath, "/") {
```

Comment:
One question: when does the `DockerSocketPath`, read from `u.Path` will not start with `/`?

Do you think this check is possible, given the docker client returns two possible values here?:
- npipe:////./pipe/docker_engine
- unix:///var/run/docker.sock

Reply:
I proceeded with caution seeing as the code in this module depends on the presence of said `/`.
For the known environments I do not see how it should miss the slash, however I think this extra if-condition is worth to save future pain from innocent users.

Reply:
OK, let's keep it, although I'd appreciate adding there a comment explaining why 🙏 

Once added, this is ready to be shipped! 🛥️

Reply:
Done 🗯️

Reply:
You are correct, I've pulled latest changes from `main` however I forgot to commit the comment @mdelapenya